tools/check.rb

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env ruby
+
+require 'yaml'
+require 'json'
+
+HUB_FED_CONFIG_DIRECTORY = File.join(__dir__, '../../verify-hub-federation-config')
+
+ENVIRONMENTS = ["prod", "integration"]
+ENVIRONMENT = ARGV[0]
+ENTITY_ID = ARGV[1]
+MSA = ARGV[2] == '--msa'
+
+if ENVIRONMENT.nil?
+  puts "USAGE: ./check.rb <environment> <entity_id> [--msa optional]"
+  exit 1
+end
+
+Dir.chdir(HUB_FED_CONFIG_DIRECTORY){
+  branch = `git rev-parse --abbrev-ref HEAD`
+  unless branch == "master\n"
+    puts "ERROR! The verify-hub-federation-config repo is not on master!"
+    exit 1
+  end
+
+  `git fetch origin master`
+
+  unless `git rev-parse HEAD` == `git rev-parse origin/master`
+    puts "ERROR! The verify-hub-federation-config repo does not appear to be in sync with origin/master. You need to git pull and retry"
+    exit 1
+  end
+}
+
+unless ENVIRONMENTS.include?(ENVIRONMENT)
+  puts "Invalid environment, accepted values are: #{ENVIRONMENTS}"
+  exit 1
+end
+
+if ENTITY_ID.nil?
+  puts "EntityID is missing"
+  exit 1
+end
+
+def compare(config)
+  entity_id = config['entityId']
+  certs = get_fed_config_certs(config)
```

Comment:
Thanks for this stuff, finally got around running the script.
Should the logic not actually be a request to the hub endpoints for encryption and signing, cos by the time we set **selfservice: true** we remove the certificates from the hub-fed-config file

Reply:
You can't hit config service from your laptop... The script should be used when onboarding the service to self-service. So after you onboarded in self-service you run the script, if happy then you go and remove the certs from the hub-fed-config